src/main/java/seedu/knowitall/logic/commands/EndCommand.java

```diff
@@ -19,11 +20,11 @@
 
     @Override
     public CommandResult execute(Model model, CommandHistory history) throws CommandException {
-        if (!model.isInTestSession() && !model.inReportDisplay()) {
+        if ((model.getState() != State.IN_TEST) && (model.getState() != State.IN_REPORT)) {
```

Comment:
Would this be better?

```suggestion
switch(model.getState()) {
    case State.IN_TEST:
        model.endTestSession();
        return new CommandResult(MESSAGE_END_TEST_SESSION_SUCCESS, CommandResult.Type.END_TEST_SESSION);
    case State.IN_REPORT:
        model.exitReportDisplay();
        return new CommandResult(MESSAGE_END_REPORT_DISPLAY_SUCCESS, CommandResult.Type.EXITED_REPORT);
    default: 
        throw new CommandException(Messages.MESSAGE_INVALID_COMMAND_OUTSIDE_FULLSCREEN);
}
```